src/ImageFitter/ImageFitter.cc

```diff
@@ -13,28 +13,38 @@
 
 using namespace carta;
 
-ImageFitter::ImageFitter(size_t width, size_t height) {
-    _fit_data.width = width;
-    _fit_data.n = width * height;
-
+ImageFitter::ImageFitter() {
     _fdf.f = FuncF;
     _fdf.df = nullptr; // internally computed using finite difference approximations of f when set to NULL
     _fdf.fvv = nullptr;
     _fdf.params = &_fit_data;
-    _fdf.n = _fit_data.n;
 
     // avoid GSL default error handler calling abort()
     gsl_set_error_handler(&ErrorHandler);
 }
 
-bool ImageFitter::FitImage(
-    float* image, const std::vector<CARTA::GaussianComponent>& initial_values, CARTA::FittingResponse& fitting_response) {
+bool ImageFitter::FitImage(size_t width, size_t height, float* image, const std::vector<CARTA::GaussianComponent>& initial_values,
+    CARTA::FittingResponse& fitting_response, size_t offset_x, size_t offset_y) {
```

Comment:
I would prefer keeping fitting_response at the end of the parameters since it is an output value but the offsets are input values.

Reply:
@pford The offset parameters are optional parameters (with default zeros), so they are added at the end.